app/models/line_item.rb

```diff
@@ -3,7 +3,9 @@ class LineItem < ApplicationRecord
   belongs_to :product
   belongs_to :cart, optional: true
   
-  validates :product_id, uniqueness: { scope: :cart_id }
+  validates :product_id, uniqueness: { scope: :cart_id }, if: :cart
+  
+  paginates_per 5
```

Comment:
`5` here should be extracted in constant in this class only

Reply:
done